src/compute/mod.rs

```diff
@@ -45,86 +48,210 @@ pub fn compute_layout(
     Ok(())
 }
 
+/// A common interface that all Taffy layout algorithms conform to
+pub(crate) trait LayoutAlgorithm {
+    /// The name of the algorithm (mainly used for debug purposes)
+    const NAME: &'static str;
+
+    /// Compute the size of the node given the specified constraints
+    fn measure_size(
+        tree: &mut impl LayoutTree,
+        node: Node,
+        known_dimensions: Size<Option<f32>>,
+        parent_size: Size<Option<f32>>,
+        available_space: Size<AvailableSpace>,
+        sizing_mode: SizingMode,
+    ) -> Size<f32>;
+
+    /// Perform a full layout on the node given the specified constraints
+    fn perform_layout(
+        tree: &mut impl LayoutTree,
+        node: Node,
+        known_dimensions: Size<Option<f32>>,
+        parent_size: Size<Option<f32>>,
+        available_space: Size<AvailableSpace>,
+        sizing_mode: SizingMode,
+    ) -> SizeAndBaselines;
+}
+
+/// The public interface to a generic algorithm that abstracts over all of Taffy's algorithms
+/// and applies the correct one based on the `Display` style
+pub struct GenericAlgorithm;
```

Comment:
Hmm. I'm not very happy with this naming / docs. Maybe something like `UnifyingLayoutAlgorithm`?

Reply:
This struct actually disappears in my latest version of #326, as I've made the "generic" algorithm and the "leaf" algorithm operate on `&mut Taffy` rather than `&mut impl LayoutTree` (so they don't fit the `LayoutAlgorithm` trait anymore). This allows us to remove measure_funcs from the `LayoutTree` trait and make them an implementation detail of our `Taffy` storage implementation.

Reply:
Good I'm fine to wait.